src/setup_node_env/index.js

```diff
@@ -17,7 +17,11 @@
  * under the License.
  */
 
-require('./harden'); // this require MUST be executed before any others
+// The following require statements MUST be executed before any others - BEGIN
+require('./exit_on_warning');
```

Comment:
do we want this in production?  i'm thinking we'll want the process to be as resilient as possible if it isn't doing any harm.

Reply:
harm is relative i guess, but warning vs error etc

Reply:
The idea is that this is disabled in production. The way this happens is that the code expects any production app to run with the `--no-warnings` flag, which you can see it does here:

https://github.com/elastic/kibana/blob/1525dbe73dda424bbc70d852b13ceb203a2d8076/bin/kibana#L24

Inside of `exit_on_warning.js` I check that `process.noProcessWarnings` is not equal to `true`, if it is, I do not attach the `warning` listener.

Reply:
awesome, thanks for the explanation